### ISSUE TITLE
RL with logits + exact rewards

### ISSUE THREAD

**Issue by tosqol** [huggingface/trl#3116]
Hello, maybe this is not the right place but any help would be appreciated!

I am trying to fine-tune a language model with a perfect reward model that takes as input the logits from my policy. There is no generation from the policy for the reward calculation, just the current policy logits on some SFT dataset. 

From my understanding this is a bit of a niche setup. Does this fit into any of the standard TRL Trainers? If not any pointers would be amazing.

Thanks!

**Comment by qgallouedec** [comment]
Sounds niche indeed, you probably want to make your own trainer from the transformers' trainer.

**Comment by tosqol** [comment]
> Sounds niche indeed, you probably want to make your own trainer from the transformers' trainer.

Got it, thanks!
